include/circt/Dialect/Moore/MooreOps.td

```diff
@@ -1006,12 +1008,21 @@ def StructExtractOp : MooreOp<"struct_extract"> {
     ```
     See IEEE 1800-2017 § 7.2.1 "Assigning to structures".
   }];
-  let arguments = (ins StrAttr:$memberName, UnpackedType:$input);
+  let arguments = (ins StrAttr:$fieldName, UnpackedType:$input);
   let results = (outs UnpackedType:$result);
   let assemblyFormat = [{
-  $input `,`   $memberName  attr-dict `:`
-  type($input) `->`
-  type($result)
+  $input `,`   $fieldName  attr-dict `:`
+  type($input) `->` type($result)
+  }];
```

Comment:
Can you add a constraint like `TypesMatchWith<...>` that ensures that the type of `$result` matches the type of the accessed field in `$input`? This will then allow you to remove the `-> type($result)` in the assembly format, since the constraint allows MLIR to determine the type automatically.

Reply:
I just add verifier for this case